docs/wip/build-pipeline.md

```diff
@@ -1,7 +1,71 @@
 ## Steps of the pipeline
+Store Framework Jamstack CI flow consists 4 main steps: `Build`, `SonarQube`, `Integration Tests` and `Lighthouse CI`. 
+
+![Store Framework Jamstack CI flow](./../images/ci_flow.png)
+
+
+This flow is triggered by two situations:
+1. When a pull request is created.
+2. When a push with commits associated to a pull request is made.
+
+When this happens, a `check suite` is created and populated with `check runs` that shows the state of each step.  
+The initial state of each step is `queued`. This indicates that soon they will be executed.
+
+![Store Framework Jamstack CI checks](./../images/queued_check_runs.png)
+
+
+## Build
+
+This step is reponsible of two main things:
+
+### Build the store
+- Run an automated build for the store, saving all the generated artifacts in our infrastructure.
+- Send the build logs to GitHub.
+The build logs are visible by clicking on `details` on the `Build` check run once its completed.
+
+![Store Framework Jamstack CI build check completed](./../images/build_check_completed.png)
+
+
+#### Possible states
+- `queued`: soon this step will be executed.
+- `in_progress`: this step is being executed.
+- `completed/success`: the build is completed successfuly.
+- `completed/failure`: the build is completed but something went wrong.
+
+### Generate a deploy preview 
+For each pull request we provide a `Deploy Preview` that is a deploy of the most recent changes of the pull request.
+You can access the `Deploy Preview` of a specific Pull Request by clicking on `details` on the commit status with the title `<your store>/deploy-preview`:  
+
+![Store Framework Jamstack CI deploy preview commit status](./../images/deploy_preview_commit_status.png)
+
+Or directly on this url: `https://preview-<pull request number>--<your store name>.vtex.app/`.  
+For example: `https://preview-417--storecomponents.vtex.app/`.
+
+
+## SonarQube
+
+SonarQube is used to mesasure and analyse code quality using static analysis.
```

Comment:
I think it would be good to refer our sonarqube documentation article here
```suggestion
[SonarQube](sonarqube.md) is used to measure and analyze code quality using static analysis.
```

Reply:
I've added a reference at the end of this section: https://github.com/vtex/faststore/pull/454/files#diff-826dc95bc82f085c38d55b2082027de2fb0ac5e9ba6d5eb96ead2cc514751a69R59

Reply:
Awesome! Could you just fix the typos in this line then?

Reply:
Great catches. Thanks!